Render constrained system friendly

ithinuel opened the pull request:
This is to allow building and running wasmi on constrained target such as microcontroller.
This requires the ability to build without std.

pepyakin commented on src/runner.rs:
```diff
@@ -18,10 +18,10 @@ use value::{
 
 /// Maximum number of levels on the call stack.
-pub const DEFAULT_CALL_STACK_LIMIT: usize = 64 * 1024;
+pub const DEFAULT_CALL_STACK_LIMIT: usize = 16 * 1024;
```
Why would you like to change the defaults? I think they are reasonable for defaults...

ithinuel replied:
The initial incentive in this change is that a lot of the device don't exceed a total of 128KB of RAM.

But that change was before I added `run_start_with_stack`. I remove this from the PR.

ithinuel commented:
@pepyakin Thanks for your review.
This is to allow building and running wasmi on constrained target such as microcontroller.
This requires the ability to build without std.
I updated the PR's description.

Since your first review I updated the truncation related changes. I removed the dependency on `BigRational`. It does not break the tests and allows to build with nostd.

ithinuel commented:
This work is also related to https://github.com/paritytech/parity-wasm/pull/283

pepyakin commented on src/nan_preserving_float.rs:
```diff
@@ -63,6 +62,9 @@ macro_rules! float {
             pub fn fract(self) -> Self {
+                #[cfg(feature = "libm")]
+                #[allow(unused_imports)]
+                use libm::{F32Ext, F64Ext};
```
Why do we need this?

pepyakin commented on src/value.rs:
```diff
@@ -369,11 +369,7 @@ macro_rules! impl_try_truncate_into {
-                    .map(|val| val.to_integer())
-                    .and_then(|val| $to_primitive(&val))
-                    .ok_or(TrapKind::InvalidConversionToInt)
+                $to_primitive(&self.trunc()).ok_or(TrapKind::InvalidConversionToInt)
```
As far I remember the current code was introduced to support 32-bit x86 platforms. Your PR is essentially reverting it here, right?

ithinuel commented:
Hi there, sorry for the way too long delay since my last update.
I ran some tests and removed the changes that are no longer required (around the float primitives).

pepyakin commented:
Thanks, @ithinuel !
